master/buildbot/www/rest.py

```diff
@@ -230,8 +230,10 @@ def writeError(msg, errcode=399,
             userinfos = self.master.www.getUserInfos(request)
             if 'anonymous' in userinfos and userinfos['anonymous']:
                 owner = "anonymous"
-            else:
+            elif userinfos['email']:
                 owner = userinfos['email']
+            else:
+                owner = userinfos['username']
```

Comment:
username is not specified as being part of the userinfos, so we should probably use .get('username')

https://docs.buildbot.net/latest/developer/cls-auth.html#buildbot.www.auth.UserInfoProviderBase

Reply:
What is best? full_name or username?
We could use email then username, then full_name or something like that.

Reply:
I think username and email are more stable and should be preferred.
looking at the implementation, it look like most userInfo providers are actually filling the username.
Perhaps best approach is to just document that we should fill the username.

Reply:
There is another source where username is mentioned: https://github.com/buildbot/buildbot/blob/master/master/docs/developer/www-server.rst#server-side-session

Reply:
I used get and looking at email, username, full_name in this order.
UserInfoProviderBase only sets email so I don't think updating alone is a good idea.